src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,240 @@
+     *
+     * To determine the available options, reference the predefined constant options in the official manual.
+     * Expected is an associative array of options types with their corresponding value.
+     *  - http://php.net/manual/en/memcached.constants.php
+     *
+     * @param string|null $dsn
+     * @param array       $opts
+     * @param string|null $persistentId
+     *
+     * @return MemcachedAdapter
+     */
+    public static function create($dsn = null, array $opts = array(), $persistentId = null)
+    {
+        if (!extension_loaded('memcached')) {
+            throw new InvalidArgumentException('Failed to create Memcache client due to missing "memcached" extension.');
+        }
+
+        $adapter = new static(new \Memcached($persistentId));
+        $adapter->setup($dsn ? array($dsn) : array(), $opts);
+
+        return $adapter;
+    }
+
+    /**
+     * Provide ability to reconfigure adapter after construction. See {@see create()} for acceptable DSN formats.
+     *
+     * @param string[] $dsns
+     * @param mixed[]  $opts
+     *
+     * @return bool
+     */
+    public function setup(array $dsns = array(), array $opts = array())
+    {
+        $ret = true;
+
+        foreach ($opts as $opt => $val) {
+            $ret = $this->setOption($opt, $val) && $ret;
+        }
+        foreach ($dsns as $dsn) {
+            $ret = $this->addServer($dsn) && $ret;
+        }
+
+        return $ret;
+    }
+
+    /**
+     * Returns the \Memcached client instance.
+     *
+     * @return \Memcached
+     */
+    public function getClient()
+    {
+        return $this->client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave(array $values, $lifetime)
+    {
+        return $this->client->setMulti($values, $lifetime) !== false
+            && $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->client->getMulti($ids) as $id => $val) {
+            yield $id => $val;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->client->get($id) !== false
+            || $this->client->getResultCode() !== \Memcached::RES_NOTFOUND;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace)
+    {
+        if (isset($namespace[0]) && false !== $ids = $this->client->getAllKeys()) {
+            return $this->doDelete(array_filter($ids, function ($id) use ($namespace) {
+                return 0 === strpos($id, $namespace);
+            }));
+        }
+
+        return $this->client->flush()
+            && $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete(array $ids)
+    {
+        $toDelete = count($ids);
+
+        foreach ((array) $this->client->deleteMulti($ids) as $result) {
```

Comment:
Older versions of the extension return a simple boolean for "all succeeded" or "one or more failures". The cast is there to allow fallback support for older versions of the extension.